### PR TITLE
Add note

### PR DESCRIPTION
**Pull request by JustinD85**
- Added 4 components
- Added redux and key for notes in redux store
- View this as something to build from, not necessary to merge

### CODE REVIEW

File: src/containers/App/App.js [hunk 1]
```diff
@@ -1,10 +1,23 @@
-import React, { Component } from 'react';
+import React, { Component } from "react";
+import { NoteArea, CreateNote } from "../";
```
Comment (dylhof) [inline src/containers/App/App.js]: This is a great way to keep our imports in control!

Comment (shannonmoranetz) [inline src/containers/App/App.js]: 💅 Great import

File: src/containers/App/App.js [hunk 2]
```diff
@@ -1,10 +1,23 @@
...
 
 class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      showPopup: false
+    };
+  }
+  handleClick = () => this.setState({ showPopup: !this.state.showPopup });
+  addNote() {}
```
Comment (dylhof) [inline src/containers/App/App.js]: Looks like this might be a left over from a refactor?

File: src/containers/App/App.js [hunk 3]
```diff
@@ -1,10 +1,23 @@
...
   render() {
+    const { showPopup } = this.state;
     return (
       <div className="App">
-        <h1 className='test'>Trapper-Keeper</h1>
+        <h1 className="title">Trapper-Keeper</h1>
+        <div onClick={this.handleClick}>Add Note</div>
+        <CreateNote canRender={showPopup} addNote={this.addNote} />
```
Comment (dylhof) [inline src/containers/App/App.js]: If the above function IS leftover, we can probably remove the addNote prop.  Let me know if this is not the case! :)

File: src/containers/NoteItemList/NoteItemList.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+
+const NoteItemList = props => (
+  <div>
+    {props.noteItemList.map(item => (
+      <div id={item.id}>{item.text}</div>
```
Comment (shannonmoranetz) [inline src/containers/NoteItemList/NoteItemList.js]: We may want to consider adding a key to these map values using UUID.

File: src/containers/CreateNote/CreateNote.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  getListItems() {
+    const { noteItemList } = this.state;
+
+    let currentList = noteItemList.map(item => (
+      <li>
+        <input id={item.id} onChange={this.handleChange} value={item.text} />
+      </li>
+    ));
+
+    currentList.push(
```
Comment (shannonmoranetz) [inline src/containers/CreateNote/CreateNote.js]: I didn't know you could push elements like this -- I LOVE this for creating our dynamic form! 👍

**Pull request merged**